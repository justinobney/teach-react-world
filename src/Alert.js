import React from 'react';

const Alert = props => (
  <div className={`alert alert-${props.type || 'info'}`} role="alert">
    {props.title && <strong>{props.title} - </strong>}
    {props.children}
  </div>
);

export default Alert;
