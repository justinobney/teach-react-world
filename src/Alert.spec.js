import Alert from './Alert';
import React from 'react';
import {create} from 'react-test-renderer';

describe('Component: Alert', () => {
  it('should render with title and children', () => {
    const tree = create(<Alert title="Title" children="Content" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow skipping title', () => {
    const tree = create(<Alert children="Content" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should default type to info', () => {
    const tree = create(<Alert children="Content" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow setting alert type', () => {
    const tree = create(<Alert children="Content" type="danger" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
