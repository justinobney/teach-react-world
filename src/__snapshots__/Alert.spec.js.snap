exports[`Component: Alert should allow setting alert type 1`] = `
<div
  className="alert alert-danger"
  role="alert">
  Content
</div>
`;

exports[`Component: Alert should allow skipping title 1`] = `
<div
  className="alert alert-info"
  role="alert">
  Content
</div>
`;

exports[`Component: Alert should default type to info 1`] = `
<div
  className="alert alert-info"
  role="alert">
  Content
</div>
`;

exports[`Component: Alert should render with title and children 1`] = `
<div
  className="alert alert-info"
  role="alert">
  <strong>
    Title
     - 
  </strong>
  Content
</div>
`;
